modules/task/backend.go

```diff
@@ -117,14 +93,39 @@ func (b *inMemBackend) Name() string {
 
 // Start implements the Module interface
 func (b *inMemBackend) Start(ready chan<- struct{}) <-chan error {
-	b.isRunning = true
-	return make(chan error)
+	if !b.IsRunning() {
+		b.isRunning = true
+		go func() {
```

Comment:
this gets pretty deeply nested, consider extracting a private method on `*inMemBackend` that you can `go` run here